src/shared/sam/localLambdaRunner.ts

```diff
@@ -79,40 +79,59 @@ export function getRelativeFunctionHandler(params: {
 }
 
 export async function makeInputTemplate(config: SamLaunchRequestArgs): Promise<string> {
-    let newTemplate = new SamTemplateGenerator()
-        .withFunctionHandler(config.handlerName)
-        .withResourceName(TEMPLATE_RESOURCE_NAME)
-        .withRuntime(config.runtime)
-        .withCodeUri(config.codeRoot)
+    let newTemplate: SamTemplateGenerator
+    let inputTemplatePath: string
 
+    // use existing template to create a temporary template with a resource that has an overridden handler name.
+    // this is necessary for Python, which overrides the handler name due to the debug file.
     if (config.invokeTarget.target === 'template') {
         const template = getTemplate(config.workspaceFolder, config)
-        // TODO: does target=code have an analog to `Globals`?
-        if (template?.Globals) {
-            newTemplate = newTemplate.withGlobals(template?.Globals)
-        }
         const templateResource = getTemplateResource(config.workspaceFolder, config)
-        if (templateResource?.Properties?.Environment?.Variables) {
-            newTemplate = newTemplate.withEnvironment({
-                Variables: templateResource?.Properties?.Environment?.Variables,
-            })
+
+        if (!template || !templateResource) {
+            throw new Error('Resource not found in base template')
         }
+
+        const resourceWithOverriddenHandler = {
+            ...templateResource,
+            Properties: {
+                ...templateResource.Properties!,
+                Handler: config.handlerName,
+            },
+        }
+
+        newTemplate = new SamTemplateGenerator(template).withTemplateResources({
+            [config.invokeTarget.logicalId]: resourceWithOverriddenHandler,
+        })
+
+        // template type uses the template dir and a throwaway template name so we can use existing relative paths
+        // clean this one up manually; we don't want to accidentally delete the workspace dir
+        inputTemplatePath = path.join(path.dirname(config.templatePath), 'app___vsctk___template.yaml')
+        // code type - generate ephemeral SAM template
     } else {
+        newTemplate = new SamTemplateGenerator()
+            .withFunctionHandler(config.handlerName)
+            .withResourceName(TEMPLATE_RESOURCE_NAME)
+            .withRuntime(config.runtime)
+            .withCodeUri(config.codeRoot)
         if (config.lambda?.environmentVariables) {
             newTemplate = newTemplate.withEnvironment({
                 Variables: config.lambda?.environmentVariables,
             })
         }
+        inputTemplatePath = path.join(config.baseBuildDir!, 'input', 'input-template.yaml')
+        // code type is fire-and-forget so we can add to disposable files
+        ExtensionDisposableFiles.getInstance().addFolder(inputTemplatePath)
```

Comment:
should it be `addFile()` ? Looks like `addFolder` works anyways, but `addFile` appears to be a safety feature of `ExtensionDisposableFiles`.

Reply:
Unsure, this is what we were using before.

Reply:
yeah but it looks like a typo. doesn't matter i guess.